src/Builder/Visitor/AddMethodCallToCallListVisitor.php

```diff
@@ -110,8 +111,15 @@ protected function isArrayMergeFuncCallNode(FuncCall $node): bool
      */
     protected function addNewCallIntoArrayMergeFuncNode(FuncCall $node, CallMetadataTransfer $callMetadataTransfer): Node
     {
-        $calledMethods = $this->getCalledMethods($node);
-        if (in_array($this->classMetadataTransfer->getTargetMethodNameOrFail(), $calledMethods)) {
+        $calledMethods = $this->getCalledMethodsFromArrayMergeFunc($node);
+        if ($this->hasTargetMethodCalled($calledMethods)) {
+            return $node;
+        }
+
+        $index = $this->classMetadataTransfer->getCall() ? $this->classMetadataTransfer->getCall()->getIndex() : null;
+        if ($index) {
```

Comment:
just if ($this->classMetadataTransfer->getCall())

Reply:
you don't use $index anywhere except of if condition, so you don't need it.

Reply:
Thanks ++